src/vos/README.md

```diff
@@ -37,80 +41,152 @@ This document contains the following sections:
+Direct access to byte-addressable low-latency storage opens up new horizons where metadata can be scanned in less than a second without bothering with seek time and alignment.
+
+The VOS relies on a log-based architecture using persistent memory primarily to maintain internal persistent metadata indexes.
+The actual data can be stored either in persistent memory directly or in block-based storage (via SPDK for instance).
+The DAOS service has two tiers of storage: Storage Class Memory (SCM) for byte-granular application data and metadata, and NVMe for bulk application data.
+Similar to how PMDK is currently used to faciliate access to SCM, the Storage Performance Development Kit (SPDK) is used to provide seamless and efficient access to NVMe SSDs.
+The current DAOS storage model involves three DAOS server xstreams per core, along with one main DAOS server xstream per core mapped to an NVMe SSD device.
+DAOS storage allocations can occur on either SCM by using a PMDK pmemobj pool, or on NVMe, using an SPDK blob.
+All local server metadata will be stored in a per-server pmemobj pool on SCM and will include all current and relevant NVMe device, pool, and xstream mapping information.
+Please refer to the Blob I/O (BIO) module for more information regarding NVMe, SPDK, and per-server metadata.
+Special care will be taken when developing the VOS layer because any software bug could corrupt data structures in persistent memory.
+The VOS therefore checksums its persistent data structures despite the presence of hardware ECC.
 
 The VOS provides a lightweight I/O stack fully in user space, leveraging the NVML open source libraries developed to support this programming model.
 
 <a id="64"></a>
 
 ### Lightweight I/O Stack: NVM Library
 
-NVML  is an open source collection of libraries for using persistent memory, optimized specifically for NVRAM. NVML is actually a collection of six libraries among which the libpmemobj library implements relocatable persistent heaps called NVML pools. This includes memory allocation, transactions, and general facilities for persistent memory programming. Locks can be embedded with PM-resident data structures, which are reinitialized (i.e. unlocked) automatically every time the NVML pool is opened. This property makes sure that all locks are always released when the NVML pool is opened.
+NVML is an open source collection of libraries for using persistent memory, optimized specifically for NVRAM.
+NVML is actually a collection of six libraries among which the libpmemobj library implements relocatable persistent heaps called NVML pools.
+This includes memory allocation, transactions, and general facilities for persistent memory programming.
+Locks can be embedded with PM-resident data structures, which are reinitialized (i.e.
+unlocked) automatically every time the NVML pool is opened.
+This property makes sure that all locks are always released when the NVML pool is opened.
```

Comment:
not sure we need to introduce NVNL lock at here, because there is no use-case in DAOS